File: Backtracking/1863. Sum of All Subset XOR Totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        def xorSUM(s, i):
            if i == len(nums):
                return s
            return xorSUM(nums[i] ^ s, i + 1) + xorSUM(s, i + 1)
        return xorSUM(0, 0)

File: Backtracking/1863. Sum of All Subset XOR Totals.cpp
class Solution {
public:
    int xorSUM(vector<int>& nums, int s, int i){
        if (i == size(nums))
            return s;
        return (xorSUM(nums, nums[i] ^ s, i + 1) + xorSUM(nums, s, i + 1));
    }
    int subsetXORSum(vector<int>& nums) {
        return xorSUM(nums, 0, 0);
    }
};